controllers/storagecluster/cephcluster.go

```diff
@@ -385,6 +381,23 @@ func validateMultusSelectors(selectors map[string]string) error {
 	return nil
 }
 
+// getNetworkSpec returns cephv1.NetworkSpec after reconciling the
+// storageCluster.Spec.HostNetwork and storageCluster.Spec.Network fields
+func getNetworkSpec(sc ocsv1.StorageCluster) cephv1.NetworkSpec {
+	// only hostnetwork specified
+	networkSpec := cephv1.NetworkSpec{HostNetwork: sc.Spec.HostNetwork}
+
+	// full NetworkSpec specified
+	if sc.Spec.Network != nil {
+		networkSpec = *sc.Spec.Network
+		// both specified
+		if sc.Spec.HostNetwork {
```

Comment:
Honestly, is this desired behavior? If a NetworkSpec even exists, I would expect we would just take it as-is, full override of anything else in the StorageCluster Spec.

Reply:
If either is set, then the user is indicating hostNetwork should be true. 
Scenario:
User sets hostNetwork, but defines NetworkSpec.IPFamily. NetworkSpec is non-nil, and NetworkSpec.hostNetwork defaults to false.

User goes: "huh, I set hostNetwork, why isn't this stupid thing turning on"